AcademicFlow.Domain.Contracts/IServices/ICourseService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using AcademicFlow.Domain.Contracts.Entities;
+
+namespace AcademicFlow.Domain.Contracts.IServices
+{
+    public interface ICourseService
+    {
+        int? AddCourse(Course course);
+
+        Course? GetCourseById(int id);
+
+        void UpdateCourse(Course course);
+
+        void DeleteCourse(int id);
+
+        IQueryable<Course> GetAll();
+        
+        void DeleteCourseUserRolesRange(IEnumerable<CourseUserRole> userRoles);
```

Comment:
why the method is not async? look at other services, it should be async and return Task